packages/react/src/lib/factories.ts

```diff
@@ -153,22 +173,28 @@ function createShorthandFromValue({
   const valIsReactElement = React.isValidElement(value)
 
   // unhandled type warning
-  if (
-    process.env.NODE_ENV !== 'production' &&
-    !valIsPrimitive &&
-    !valIsPropsObject &&
-    !valIsArray &&
-    !valIsReactElement &&
-    !valIsNoop
-  ) {
-    /* eslint-disable-next-line no-console */
-    console.error(
-      [
-        'Shorthand value must be a string|number|object|array|ReactElements.',
-        ' Use null|undefined|boolean for none.',
-        ` Received: ${value}`,
-      ].join(''),
-    )
+  if (process.env.NODE_ENV !== 'production') {
+    if (!valIsPrimitive && !valIsPropsObject && !valIsArray && !valIsReactElement && !valIsNoop) {
+      /* eslint-disable-next-line no-console */
+      console.error(
+        [
+          'Shorthand value must be a string|number|object|array|ReactElements.',
+          ' Use null|undefined|boolean for none.',
+          ` Received: ${value}`,
+        ].join(''),
+      )
+    }
+
+    if (!allowsJSX && valIsReactElement) {
+      /* eslint-disable-next-line no-console */
+      console.error(
+        [
+          'Shorthand value must be a string|number|object|array when it is a `allowsJSX={false}`.',
+          ' Use null|undefined|boolean for none.',
+          ` Received: ${value}`,
```

Comment:
Invoke this error in the browser and make sure it makes good sense to the user.  Probably would be a good idea to show the component name, prop name, and value in the error.  Something like:

```
The shorthand prop `{prop}` was passed a JSX element but this slot only supports string|number|object|array
```

This is a rough idea, just make sure user's will be well informed and happy 👍

Reply:
`The shorthand prop for "${Component.displayName}" component was passed a JSX element but this slot only supports string|number|object|array|ReactElements.`,

The best that I was able to do because factories don't know about parent components 😭